client/src/components/ProfileForm.tsx

```diff
@@ -0,0 +1,173 @@
+import { useState } from "react";
+import {
+  Button,
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalCloseButton,
+  ModalBody,
+  ModalFooter,
+  Input,
+  Textarea,
+  useToast,
+} from "@chakra-ui/react";
+import { Profile } from "types/Profile";
+import { useSelector } from "react-redux";
+import { userSelector } from "redux/user";
+
+interface UserData {
+  displayName: string;
+  email: string;
+}
+
+export const ProfileForm = (props: {
+  isOpen: boolean;
+  onClose: () => void;
+  isEditing: boolean;
+  setIsEditing: React.Dispatch<React.SetStateAction<boolean>>;
+  userData: UserData | null | undefined;
+  profileData: Profile;
+}) => {
+  const API_URL = `${import.meta.env.VITE_API_SERVER_URL}/api/v1`;
+  const AuthToken = localStorage.getItem("BEARER_TOKEN") || "";
```

Comment:
move api url outside of the function

don't use localStorage, you already using userSelector below